src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -199,8 +200,11 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
                 },
                 ThreadHelper.JoinableTaskFactory);
 
-            AddService(typeof(ExternalSettingsProviderService),
-                (container, ct, serviceType) => Task.FromResult<object>(new ExternalSettingsProviderService()),
+            AddService(typeof(GeneralExternalSettingsProviderService),
+                (container, ct, serviceType) => Task.FromResult<object>(new GeneralExternalSettingsProviderService()),
+                promote: true);
+            AddService(typeof(ConfigurationFilesExternalSettingsProviderService),
+                (container, ct, serviceType) => Task.FromResult<object>(new ConfigurationFilesExternalSettingsProviderService()),
```

Comment:
Would it be possible to pass the Settings as a construction parameter? That would avoid having to use the service locator and would make the dependencies of the ConfigurationFilesExternalSettingsProviderService more clearly defined in the constructor.

Reply:
Good idea! Done.